openshift-knative-operator/pkg/monitoring/monitoring.go

```diff
@@ -200,6 +282,66 @@ func createServiceMonitorService(component string) corev1.Service {
 				Name: "https",
 				Port: 8444,
 			}},
-			Selector: map[string]string{"app": component},
+			Selector: getSelectorLabels(component),
 		}}
 }
+
+func createClusterRoleBindingManifest(serviceAccountName string, ns string) (*mf.Manifest, error) {
+	crb := v1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("rbac-proxy-reviews-prom-rb-%s", serviceAccountName),
+		},
+		RoleRef: v1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "rbac-proxy-reviews-prom",
+		},
+		Subjects: []v1.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      serviceAccountName,
+			Namespace: ns,
+		}},
+	}
+	var crbU = &unstructured.Unstructured{}
+	if err := scheme.Scheme.Convert(&crb, crbU, nil); err != nil {
+		return nil, err
+	}
+	manifest, err := mf.ManifestFrom(mf.Slice([]unstructured.Unstructured{*crbU}))
+	if err != nil {
+		return nil, err
+	}
+	return &manifest, nil
+}
+
+// getDefaultMetricsPort returns the expected metrics port under the assumption that this will not change
+// This is static information since observability cm does not allow any changes for the prometheus config
+func getDefaultMetricsPort(name string) string {
+	if servingComponents.Has(name) {
+		return "9090"
+	}
+	if name == "mt-broker-ingress" || name == "mt-broker-filter" {
+		return "9092"
+	}
+	return "9090"
+}
+
+func getSelectorLabels(component string) map[string]string {
```

Comment:
perhaps a comment on the exceptions/cases?

Reply:
done